test/distributed/_shard/sharded_tensor/test_sharded_tensor.py

```diff
@@ -1787,32 +1784,22 @@ def test_with_rpc_names(self):
 
 class TestShardedTensorFromLocalTensor(ShardedTensorTestBase):
     def _generate_st_from_chunk_local_tensor(self, st_size, sharding_spec):
-        chunk_sharding_dim = sharding_spec.dim
-        world_size = len(sharding_spec.placements)
-        split_size = get_split_size(st_size[chunk_sharding_dim], world_size)
-        # TODO: To use an easier way to generate the ShardedMetadatas
-        # for sharding spec after https://github.com/pytorch/pytorch/pull/72130.
-        shards_metadata = [None] * world_size
-        local_shard_meta = None
-        for idx, placement in enumerate(sharding_spec.placements):
-            shard_size = copy.deepcopy(st_size)
-            offsets = [0] * len(st_size)
-            offsets[chunk_sharding_dim] = min(
-                split_size * idx, st_size[chunk_sharding_dim]
-            )
-            shard_size[chunk_sharding_dim] = get_chunked_dim_size(
-                st_size[chunk_sharding_dim], split_size, idx
-            )
-            shards_metadata[placement.rank()] = ShardMetadata(
-                shard_offsets=offsets,
-                shard_sizes=shard_size,
-                placement=placement,
-            )
-            if placement.rank() == self.rank:
-                local_tensor_size = copy.deepcopy(shard_size)
-                local_shard_metadata = shards_metadata[placement.rank()]
+        tensor_meta = sharding_spec.build_metadata(st_size, TensorProperties())
+        pg = dist.distributed_c10d._get_default_group()
 
-        local_tensor = torch.rand(*local_tensor_size).cuda(self.rank)
+        local_tensor = None
+        local_shard_metadata = None
+        rank_to_metadata = {}
+        for shard_metadata in tensor_meta.shards_metadata:
+            rank, device = _parse_and_validate_remote_device(pg, shard_metadata.placement)
+            rank_to_metadata[rank] = shard_metadata
+            if rank == self.rank:
+                local_tensor = torch.rand(shard_metadata.shard_sizes).cuda(device)
+                local_shard_metadata = shard_metadata
+
+        # TODO: figure out what the API should behave when some rank have no shard
```

Comment:
made some changes in test about `init_from_local_tensor`, to reflect our offline discussion

Reply:
sounds good to me.